server/config/config.go

```diff
@@ -112,23 +120,25 @@ func (conf *Config) Update(stored StoredConfig, mmconf *model.Config, license *m
 			host = "127.0.0.1"
 		}
 
-		localURL = "http://" + host + ":" + port + mattermostURL.Path
+		localURL = "http://" + host + ":" + port + u.Path
 	}
 
 	conf.StoredConfig = stored
 
-	conf.MattermostSiteURL = *mattermostSiteURL
-	conf.MattermostSiteHostname = mattermostURL.Hostname()
+	conf.MattermostSiteURL = u.String()
 	conf.MattermostLocalURL = localURL
 	conf.PluginURLPath = "/plugins/" + conf.PluginManifest.Id
-	conf.PluginURL = strings.TrimRight(*mattermostSiteURL, "/") + conf.PluginURLPath
+	conf.PluginURL = strings.TrimRight(u.String(), "/") + conf.PluginURLPath
 
 	conf.MaxWebhookSize = 75 * 1024 * 1024 // 75Mb
 	if mmconf.FileSettings.MaxFileSize != nil {
 		conf.MaxWebhookSize = *mmconf.FileSettings.MaxFileSize
 	}
 
 	conf.DeveloperMode = pluginapi.IsConfiguredForDevelopment(mmconf)
+	if devHostRegexp.MatchString(u.Hostname()) {
+		conf.DeveloperMode = true
+	}
```

Comment:
`DeveloperMode` is also the _unsafe_ mode. I'm concerned about using that mode in somewhat production instance. At least community is one.

Is there another approach that we can take? Like enabling `http` Apps on Cloud servers?

Reply:
+1 for just enabling http instead of dev mode

Reply:
I will introduce a separate flag for "allow install HTTP".